contracts/modules/v2/Assessment.sol

```diff
@@ -0,0 +1,754 @@
+          payoutStatusDisplay = "Complete";
+        } else {
+          payoutStatusDisplay = "Pending";
+        }
+      }
+      // [todo] Get from covers contract
+      uint coverStart = block.timestamp;
+      uint coverPeriod = 365;
+      uint coverEnd = coverStart + coverPeriod * 1 days;
+      uint productId = 1;
+      uint voteEnd = getVotingPeriodEnd(EventType.CLAIM, claimId);
+      string memory assetSymbol;
+      {
+        if (claim.details.asset == Asset.ETH) {
+          assetSymbol = "ETH";
+        } else if (claim.details.asset == Asset.DAI) {
+          assetSymbol = "DAI";
+        }
+      }
+      claimDisplays[claimId - from] = ClaimDisplay(
+        claimId,
+        productId,
+        claim.details.coverId,
+        claim.details.amount,
+        assetSymbol,
+        coverStart,
+        coverEnd,
+        claim.poll.voteStart,
+        voteEnd,
+        claimStatusDisplay,
+        payoutStatusDisplay
+      );
+    }
+    return claimDisplays;
+  }
+
+  function getSubmissionFee() internal view returns (uint) {
+    return 1 ether * uint(FLAT_ETH_FEE_PERC) / uint(PERC_BASIS_POINTS);
+  }
+
+  function getVoteCountOfAssessor(address assessor) external view returns (uint) {
+    return votesOf[assessor].length;
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  function _setClaimPayoutComplete (ClaimDetails storage details) internal {
+    // One way operation.
+    uint16 flipPosition = 1 << 15;
+    details.misc = details.misc | flipPosition;
+  }
+
+  /**
+   *  Submits a claim for assessment
+   *
+   *  @dev This function requires an ETH submission fee. See: getSubmissionFee()
+   *
+   *  @param coverId          Cover identifier
+   *  @param requestedAmount  The amount expected to be received at payout
+   *  @param withProof        When true, a ProofSubmitted event is emitted with ipfsProofHash.
+   *                          When false, no ProofSubmitted event is emitted to save gas if the
+   *                          cover wording doesn't enforce a proof of loss.
+   *  @param ipfsProofHash    The IPFS hash required for proof of loss. It is ignored if withProof
+   *                          is false
+   */
+  function submitClaimForAssessment(
+    uint24 coverId,
+    uint104 requestedAmount,
+    bool withProof,
+    string calldata ipfsProofHash
+  ) external payable onlyMember {
+    require(
+      msg.value == getSubmissionFee(),
+      "Assessment: Submission fee different that the expected value"
+     );
+    // [todo] Cover premium and total amount need to be obtained from the cover
+    // itself. The premium needs to be converted to NXM using a TWAP at claim time.
+    uint104 coverAmount = 1000 ether;
+    uint16 coverPeriod = 365;
+    Asset asset = Asset.ETH; // take this form cover asset
+    uint80 nxmPriceSnapshot = uint80(1 ether);
+    require(coverPeriod <= 2**11-1, "Assessment: Cover period cannot exceed 2047 days");
```

Comment:
this should be enforced at cover level, not here. 
max cover period will be 1 year.